agent/eventhandler/task_handler.go

```diff
@@ -15,63 +15,80 @@ package eventhandler
 
 import (
 	"container/list"
+	"sync"
 	"time"
 
 	"github.com/aws/amazon-ecs-agent/agent/api"
 	"github.com/aws/amazon-ecs-agent/agent/utils"
 )
 
-var handler *taskHandler
+// Maximum number of tasks that may be handled at once by the TaskHandler
+const concurrentEventCalls = 3
 
-func init() {
-	handler = newTaskHandler()
+type eventList struct {
+	*list.List            // list of *sendableEvents
+	sending    bool       // whether the list is already being handled
+	mu         sync.Mutex // Locks both the list and sending bool
```

Comment:
Please use a more descriptive name for this: `lock` or better yet specify what this lock is supposed to protect. `listLock` for example.